sklearn/model_selection/tests/test_split.py

```diff
@@ -1901,3 +1928,23 @@ def test_random_state_shuffle_false(Klass):
 )
 def test_yields_constant_splits(cv, expected):
     assert _yields_constant_splits(cv) == expected
+
+
+@pytest.mark.parametrize("cv", ALL_SPLITTERS, ids=[str(cv) for cv in ALL_SPLITTERS])
+def test_splitter_get_metadata_routing(cv):
+    assert hasattr(cv, "get_metadata_routing")
+    if cv in GROUP_SPLITTERS:
+        assert (
+            str(cv.get_metadata_routing())
+            == "{'split': {'groups': <RequestType.REQUESTED: True>}}"
```

Comment:
Testing the string representation seems a little fragile. Can we test the metadata itself?

Reply:
I was thinking to test the object directly:

```python
    metadata = cv.get_metadata_routing()
    if cv in GROUP_SPLITTERS:
        assert metadata.split.requests["groups"] == RequestType.REQUESTED
    elif cv in NO_GROUP_SPLITTERS:
        assert not metadata.split.requests
```

Although this test is weaker than the correct one. I do not see a simple way to test "no other routes set" besides:

```python
    assert all(
        not route.requests for key, route in vars(metadata).items() if key != "split"
    )
```

I guess the string representation is the simplest one for this type of test.

Reply:
That's why I have these in `test_metadata_routing.py`, but I'm not super happy with them:

https://github.com/scikit-learn/scikit-learn/blob/0b298edc5bd0333a7501738dcd50f2ea2cd83ee1/sklearn/tests/test_metadata_routing.py#L38-L70

Reply:
Ooo we can use that here. What about:

```python
from sklearn.utils.metadata_routing import RequestType
from sklearn.tests.test_metadata_routing import assert_request_is_empty

def test_splitter_get_metadata_routing(cv):
    assert hasattr(cv, "get_metadata_routing")
    metadata = cv.get_metadata_routing()
    if cv in GROUP_SPLITTERS:
        assert metadata.split.requests["groups"] == RequestType.REQUESTED
    elif cv in NO_GROUP_SPLITTERS:
        assert not metadata.split.requests

    assert_request_is_empty(metadata, exclude=["split"])
```

it's less brittle than string testing.